app/models/satisfaction_survey_feature_metrics.rb

```diff
@@ -42,6 +60,16 @@ def application_forms_with_feedback(start_time, end_time)
       .where(recruitment_cycle_year: RecruitmentCycle.current_year)
   end
 
+  def application_forms_with_positive_feedback(start_time, end_time)
+    ApplicationForm
+      .where(
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+      )
+      .where('application_forms.submitted_at BETWEEN ? AND ?', start_time, end_time)
+      .where(feedback_satisfaction_level: 'very_satisfied').or(ApplicationForm.where(feedback_satisfaction_level: 'satisfied'))
+      .where(recruitment_cycle_year: RecruitmentCycle.current_year)
```

Comment:
I don't think this query is quite right, it gives me the following SQL (with some sample dates):
```
SELECT application_forms.*
FROM application_forms
WHERE (application_forms.recruitment_cycle_year = 2021 AND
  (application_forms.submitted_at BETWEEN '2021-06-30 12:11:42.072318' AND '2021-07-08 12:11:42.072536')
  AND application_forms.feedback_satisfaction_level = 'very_satisfied'
  OR application_forms.feedback_satisfaction_level = 'satisfied')
AND application_forms.recruitment_cycle_year = 2021;
```

which is logically...

```
SELECT application_forms.*
FROM application_forms
WHERE (application_forms.recruitment_cycle_year = 2021 AND
  ((application_forms.submitted_at BETWEEN '2021-06-30 12:11:42.072318' AND '2021-07-08 12:11:42.072536')
  AND application_forms.feedback_satisfaction_level = 'very_satisfied')
  OR application_forms.feedback_satisfaction_level = 'satisfied')
AND application_forms.recruitment_cycle_year = 2021;
```

so you can end up with satisfaction rates > 100%

Reply:
I think you need something like this...

```
    ApplicationForm
      .where(feedback_satisfaction_level: 'very_satisfied').or(ApplicationForm.where(feedback_satisfaction_level: 'satisfied'))
      .where(recruitment_cycle_year: RecruitmentCycle.current_year)
      .where('application_forms.submitted_at BETWEEN ? AND ?', start_time, end_time)
      .where(recruitment_cycle_year: RecruitmentCycle.current_year)
```

(the bracketing around `OR` that you end up with isn't very obvious - at least not to me).

Reply:
or better still use an `IN` as @frankieroberto suggests

Reply:
okay cool so something like this @stevehook @frankieroberto 

`  def application_forms_with_positive_feedback(start_time, end_time)
    ApplicationForm
      .where(feedback_satisfaction_level: %w[very_satisfied satisfied])
      .where(recruitment_cycle_year: RecruitmentCycle.current_year)
      .where(application_forms: { submitted_at: start_time..end_time })
      .where(recruitment_cycle_year: RecruitmentCycle.current_year)
  end`

Reply:
@george-bit  Looks good to me, although there's a duplicate line. Depending on preferred code style, could even combine the `.where` statements:

```ruby
def application_forms_with_positive_feedback(start_time, end_time)
    ApplicationForm
      .where(
        feedback_satisfaction_level: %w[very_satisfied satisfied],
        recruitment_cycle_year: RecruitmentCycle.current_year,
        submitted_at: start_time..end_time
      )
  end
  ```

Reply:
This is much neater thanks.